### PR TITLE
Earthdata auth errors

### PR DESCRIPTION
**Pull request by jtherrmann**
See https://github.com/ASFHyP3/hyp3-sdk/issues/170 and https://github.com/ASFHyP3/hyp3-sdk/pull/194

TODO:

- [x] Fix the broken test
- [x] Add more tests for remaining test cases

### CODE REVIEW

File: src/hyp3_sdk/config.py
Comment (forrestfwilliams) [inline src/hyp3_sdk/config.py:1]: Great to see use moving to mocking over this.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [PEP 440](https://www.python.org/dev/peps/pep-0440/)
 and uses [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 
+## [1.8.0]
+### Changed
+* Improved error messages when Earthdata user must select Study Area or accept EULA.
+
```
Comment (forrestfwilliams) [inline CHANGELOG.md:13]: I'd like to see a note here about how we've updated our testing approach to use mocks.

File: src/hyp3_sdk/hyp3.py
Comment (forrestfwilliams) [inline src/hyp3_sdk/hyp3.py:1]: Are these changes necessary for the new functionality, or are they just a bit of clean-up? Either way, I like them.

File: CHANGELOG.md (hunk 1 above)
Comment (jtherrmann) [reply CHANGELOG.md:13]: Hmm, I've usually avoided putting stuff in the Changelog that's related to refactoring or "internal" changes that users won't care about. There's some interesting discussion on this point [here](https://github.com/olivierlacan/keep-a-changelog/discussions/463).

However, in this case, it actually does change our public API because the `hyp3_sdk.TESTING` global variable no longer exists. Technically I think that means this should be a major release version, but I really hope no users were relying on that variable.

Anyway, I will go ahead and add it to the Changelog.

File: src/hyp3_sdk/hyp3.py
Comment (jtherrmann) [reply src/hyp3_sdk/hyp3.py:1]: It was necessary to `import hyp3_sdk.util` so that I could patch `hyp3_sdk.util.get_authenticated_session` in the unit tests. The `patch` function is very particular about how objects are referenced, so it wouldn't work with `from hyp3_sdk.util import get_authenticated_session`.

Comment (jtherrmann) [reply src/hyp3_sdk/hyp3.py:1]: I don't really understand all the details of how `patch` resolves object paths, so I find it's easiest to just avoid `from X import Y` statements for any objects that must be patched.

**Review by forrestfwilliams — APPROVED** [review]
